tests/auth_tests/test_tokens.py

```diff
@@ -140,3 +140,48 @@ def test_secret_lazy_validation(self):
         msg = 'The SECRET_KEY setting must not be empty.'
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
             default_token_generator.secret
+
+    def test_verify_with_secret_fallbacks(self):
+        user = User.objects.create_user('tokentestuser', 'test2@example.com', 'testpw')
+
+        p1 = PasswordResetTokenGenerator()
+        p1.secret = 'secret'
+
+        p2 = PasswordResetTokenGenerator()
+        p2.secret = 'newsecret'
+        p2.secret_fallbacks = ['secret']
+
+        tk1 = p1.make_token(user)
```

Comment:
For all these tests, I'd move `tk1 = p1.make_token(user)` up before instantiating `p2`, so that `p1` drops out of scope. 

I suspect @felixxm would like you to remove (or at least reduce) the whitespace.

Reply:
@carltongibson I'll make those changes, but when you have time can you clarify what you mean by `p1` drops out of scope? I believe you mean it's limited to those few lines. However, I would use the term `scope` to describe anything in a block of code such as a function or if statement. And I can't remove `p1` because `secret` isn't a parameter to `PasswordResetTokenGenerator` which means `p1` is still going to remain in scope (in the technical sense) for the whole test.

It's not a big deal, but looking to understand.

Reply:
I just mean the number of line it's _active_ for. Bar the assert in the case, `p1` is just used to create `tk1` and then not mentioned again (if I read correctly this morning ☕) If you pull up the creation of `tk1` then we never have to consider `p1` again. 

The idea is to reduce the time the variable is "live" — looking it up, see Code Complete Ch10 if you have that. A google gives this http://www.rosscode.com/blog/index.php?title=variable_usage_span_and_live_time&more=1&c=1&tb=1&pb=1 on the topic.